bson/src/main/org/bson/internal/UnsignedInts.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ * Copyright 2010 The Guava Authors
+ * Copyright 2011 The Guava Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.internal;
+
+import java.math.BigInteger;
+
+/**
+ * Utilities for treating int values as unsigned.
+ *
+ * <p>
+ * Similar methods are now available in Java 8, but are required here for Java 6/7 compatibility.
+ * </p>
+ * <p>
+ * This class is not part of the public API and may be removed or changed at any time.
+ * </p>
+ */
+public final class UnsignedInts {
```

Comment:
I don't think this class is necessary given what ObjectId actually requires.  See my comment below.